src/replication_core.rs

```diff
@@ -20,52 +19,90 @@ impl Plugin for ReplicationCorePlugin {
     }
 }
 
+/// Event delivery guarantee.
+#[derive(Clone, Copy)]
```

Comment:
One more thing. Could you put this enum and `DEFAULT_EVENT_CHANNEL_RESEND_TIME` to `network_event` since it relevant only for it?